publish-to-npm.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
```

Comment:
this script is small and readable so far, but it won't stay like this over time I am afraid, even now I would rewrite it in ts, so we all understand what it does clearly

I personally am a complete bash noob.

Another point is if this was merged, what stops us from merging another bash script later? and another one?

Reply:
I'm bash noob as well but simple scripts can be more expressive than ts. Since ts is not  shell language. And everything googleable.

> what stops us from merging another bash script later?

If those will be simple scripts I wouldn't worry.
This one is specifically for github actions which are bash as well by default.